src/jekyll/jk_document.rs

```diff
@@ -0,0 +1,219 @@
+use liquid;
+
+use super::super::frontmatter;
+use super::super::datetime;
+use std::fs::{File, create_dir_all};
+use std::io::Write;
+use std::path::Path;
+use super::super::document::{read_file, split_document};
+use frontmatter::FrontmatterBuilder;
+use error::{ErrorKind, Result};
+use serde_yaml;
+
+#[derive(Debug, Eq, PartialEq, Default, Clone)]
+#[derive(Serialize, Deserialize)]
+pub struct JkFrontmatterBuilder(liquid::Object);
+
+impl JkFrontmatterBuilder {
+    pub fn new() -> JkFrontmatterBuilder {
+        JkFrontmatterBuilder(liquid::Object::new())
+    }
+}
+
+impl From<JkFrontmatterBuilder> for frontmatter::FrontmatterBuilder {
+    fn from(jk_front: JkFrontmatterBuilder) -> Self {
+        // Convert jekyll frontmatter into frontmatter (with `custom`)
+        let mut custom_attributes = jk_front.0;
+        frontmatter::FrontmatterBuilder::new()
+            .merge_title(custom_attributes.remove("title").and_then(|v| {
+                v.as_str().map(|s| s.to_owned())
+            }))
+            .merge_slug(custom_attributes.remove("permalink").and_then(|v| {
+                v.as_str().map(|s| s.to_owned())
+            }))
+            .merge_description(custom_attributes.remove("excerpt").and_then(|v| {
+                v.as_str().map(|s| s.to_owned())
+            }))
+            .merge_categories(custom_attributes.remove("categories").and_then(|v| {
+                v.as_array().map(
+                    |v| v.iter().map(|v| v.to_string()).collect(),
+                )
+            }))
+            .merge_permalink(custom_attributes.remove("permalink").and_then(|v| {
```

Comment:
Does the permalink format need converting?

Even if it doesn't need it now, we should ensure there are tests for the variable substitution because it might change in the future and we need to make sure we know to update this code in that case.

Reply:
After reading "Custom Path" on http://cobalt-org.github.io/docs/pages.html it seems that the plan was to copy Jekyll behavior, so no conversion needed.

Reply:
At first, I thought this project was meant to be a Jekyll clone.  Later I found out its more Jekyll inspired.  I've not yet updated that page.  I have made some minor tweaks to #198 to reflect that.

One example change I want to make: if people think its reasonable, I'd like to move away from `:var` substitutions.  The grammar is ambiguous (when is the end of a `var`?).  Currently, we handle this by enumerating all the supported vars and doing string substitutions.  We don't have the opportunity to detect unsupported vars.

One idea I had was to replace the system with Liquid.

Reply:
ok, I kinda understand the concept here, my knowledge on Jekyll and Cobalt is not advanced enough to fully grasp the issue but I can add a function call in the conversion here so if some conversions are needed, we already have the entry point to do so